api.planx.uk/modules/analytics/metabase/shared/client.test.ts

```diff
@@ -1 +1,127 @@
-test.todo("should test configuration and errors");
+import axios from "axios";
+import {
+  validateConfig,
+  createMetabaseClient,
+  MetabaseError,
+  initializeMetabaseClient,
+} from "./client.js";
+import nock from "nock";
+
+const axiosCreateSpy = vi.spyOn(axios, "create");
+
+describe("Metabase client", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.resetModules();
+    process.env.METABASE_URL_EXT = "https://test-metabase-url.com";
+    process.env.METABASE_API_KEY = "test-api-key";
+    initializeMetabaseClient();
+  });
+
+  afterEach(() => {
+    vi.unstubAllEnvs();
+  });
+
+  test("returns configured client", async () => {
+    const _client = createMetabaseClient();
+
+    expect(axiosCreateSpy).toHaveBeenCalledWith({
+      baseURL: process.env.METABASE_URL_EXT,
+      headers: {
+        "X-API-Key": process.env.METABASE_API_KEY,
+        "Content-Type": "application/json",
+      },
+      timeout: 30_000,
+    });
+  });
+
+  describe("validates configuration", () => {
+    test("throws error when URL_EXT is missing", () => {
+      vi.stubEnv("METABASE_URL_EXT", undefined);
+      expect(() => validateConfig()).toThrow(
+        "Missing environment variable 'METABASE_URL_EXT'",
+      );
+    });
+
+    test("throws error when API_KEY is missing", () => {
+      vi.stubEnv("METABASE_API_KEY", undefined);
+      expect(() => validateConfig()).toThrow(
+        "Missing environment variable 'METABASE_API_KEY'",
+      );
+    });
+
+    test("returns valid config object", () => {
+      const config = validateConfig();
+      expect(config).toMatchObject({
+        baseURL: process.env.METABASE_URL_EXT,
+        apiKey: process.env.METABASE_API_KEY,
+        timeout: 30_000,
+        retries: 3,
+      });
+    });
+  });
+
+  describe("Error handling", () => {
+    test("retries then succeeds on 5xx errors", async () => {
+      const baseURL = process.env.METABASE_URL_EXT;
+      if (!baseURL) {
+        throw new Error("METABASE_URL_EXT must be defined for tests");
+      }
+
+      const metabaseScope = nock(baseURL);
```

Comment:
It's actually in there 'cause I get a type error if I try another method, eg just going with `const metabaseScope = nock(process.env.METABASE_URL_EXT);` 
results in this error, which I haven't figured out a decent way of addressing:
``` Argument of type 'string | undefined' is not assignable to parameter of type 'string | RegExp | Url | URL'.
  Type 'undefined' is not assignable to type 'string | RegExp | Url | URL'.
  ```

Reply:
This works ok for me using `!` ([docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator))

<img width="537" alt="image" src="https://github.com/user-attachments/assets/0c3e3a0f-0ceb-422d-88fc-cec88a2d20a3">

Is this working locally but failing on CI? If so that might indicate an issue with the secrets on S3 and not the code itself. Happy to try and solve this on a call tomorrow 👍